### DISCUSSION TITLE
Is the "Tabs" permission really required?

### DISCUSSION THREAD

**Discussion by GraphiteEater** [KristofferTroncoso/ThemeSong#5] [General]
I don't think so...

Also the permission warning from Chrome is scary: "Read your browsing history".
If there's a way to not use the "Tabs" permission but still keep the same UI/UX, that would be nice.

**Comment by KristofferTroncoso** [comment]
True true. You make a good point GraphiteEater.
It was required just to execute and update the content script onto https://music.youtube.com tabs. I definitely don't care about someone else's browsing history haha.

I'll see if i can do it another way while keeping the same seamless UX and also DX.

**Comment by KristofferTroncoso** [comment]
There's a lot i can try here:
https://developer.chrome.com/docs/extensions/reference/scripting/
The updateContentScripts method is important too (https://developer.chrome.com/docs/extensions/reference/scripting/#method-updateContentScripts) since it needs to seamlessly update the contentScript on extension update. 
If it doesn't update correctly, the Popup settings loses context with the contentScript which is basically "broken".

**Comment by KristofferTroncoso** [comment]
Uh... this is embarassing...
I found out I don't even need the "Tabs" permission to use the `chrome.tabs.query()` method. 
"Tabs" is only needed if I pass in the url key like so `chrome.tabs.query({url: "https://music.youtube.com"})`.
If I pass in an empty object `chrome.tabs.query({})`, as long as I have "https://music.youtube.com*" in my host_permissions in my manifest, chrome.tabs.query() will tell me which ones are YouTube Music tabs. All other tabs will remain anonymous.

This is the best solution for now since it takes away the "Tabs" permission while keeping the same flow.
I can research and explore other methods like dynamic scripts and unregistering scripts some other time.
